lib/utils/setCookie.js

```diff
@@ -55,14 +56,17 @@ module.exports = function setCookie (storage, name, value) {
 
       for (; i < ll; i++) {
         clone.domain = domains[i]
-        cookie.setItem(name, value, expires, clone.path, clone.domain, true, 'None')
 
-        // Break when cookies aren't being cleared and it gets set properly
-        // Don't break when value is falsy so all the cookies get cleared
-        if (cookie.getItem(name) === value) {
-          // When cookie is set succesfully, save used domain in storage object
-          storage.domain = clone.domain
-          break
+        if (isValidDomain(clone.domain)) {
```

Comment:
Maybe we can put this `isValidDomain` inside `findDomains` function?
- **Pro** is that this makes more correct `findDomains`
- **Cons** is that this takes more computation for this case because we don't have benefit of `break` argument anymore